File: 0525-contiguous-array/0525-contiguous-array.cpp
class Solution {
public:
    int findMaxLength(vector<int>& nums) {
        
            int sum=0;
            int max_length=0;
            unordered_map<int,int>gzip;
            gzip[0]=-1;
           for(int i=0;i<nums.size();i++){
                        
                    sum+=nums[i]==0?-1:1;
                    
                    if(gzip.find(sum)!=gzip.end()){
                        
                             max_length=max(max_length,i-gzip[sum]);   
                        
                    }
                    else{
               
                        gzip[sum]=i; 
                    }
            }
        
            return max_length;
        
    }
};

File: 0525-contiguous-array/0525-contiguous-array.py
class Solution:
    
    def findMaxLength(self, nums: List[int]) -> int:
            
            
                max_len=0
                total=0
                dicti={}
                j=0
                dicti[0]=-1
                while(j<len(nums)):
                    
                    if nums[j]==1:
                        total+=1
                    else:
                        total-=1

                    if total not in dicti:
                            dicti[total]=j
                    else:
                            max_len=max(max_len,j-dicti[total])
                    
                    j+=1
                        
                            
                return max_len            
                                
                                
                    
                        
                    